### PR TITLE
Add Q3 2020 roadmap

### PR DESCRIPTION
**Pull request by Redeux**
### Acceptance tests
- [ ] Have you added an acceptance test for the functionality being added?
- [ ] Have you run the acceptance tests on this branch? (If so, please include the test log in a gist)

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

File: _about/ROADMAP.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+### Kubernetes Manifest Resource
+
+Issue: [#215](https://github.com/hashicorp/terraform-provider-kubernetes/issues/215)
+
+We’ve been evaluating the best way to support custom resources in the Kubernetes provider. The [initial work](https://github.com/hashicorp/terraform-provider-kubernetes-alpha) led us to create a new provider with a single resource, called `kubernetes_manifest`. We did this because the functionallity needed to accomidate the dynamic schema of custom resources wasn't available in the Terraform SDK.  
+
+Now that the features we need are planned for a near-term SDK release, we're evaluating the feasability of merging the Kubernetes-alpha manifest resource into the existing Kubernetes provider.
+
+We are interested in your thoughts and feedback about this proposal and encourage you to comment on the issue linked above or schedule time with @redeux via the link on his [GitHub profile](https://github.com/redeux) to discuss.
```
Comment (aareet) [inline _about/ROADMAP.md:60]: Maybe put the calendly link here to avoid the redirect?

File: _about/ROADMAP.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+Research topics include features, architectural changes, and ideas that we are pursuing in the longer term that may significantly impact the core user experience of the Kubernetes provider. Research topics are discovery only and are not guaranteed to be included in a future release.
+
+### Kubernetes Manifest Resource
+
+Issue: [#215](https://github.com/hashicorp/terraform-provider-kubernetes/issues/215)
+
+We’ve been evaluating the best way to support custom resources in the Kubernetes provider. The [initial work](https://github.com/hashicorp/terraform-provider-kubernetes-alpha) led us to create a new provider with a single resource, called `kubernetes_manifest`. We did this because the functionallity needed to accomidate the dynamic schema of custom resources wasn't available in the Terraform SDK.  
+
+Now that the features we need are planned for a near-term SDK release, we're evaluating the feasability of merging the Kubernetes-alpha manifest resource into the existing Kubernetes provider.
```
Comment (aareet) [inline _about/ROADMAP.md:58]: typo - `feasibility`

File: _about/ROADMAP.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+## Research Topics
+
+Research topics include features, architectural changes, and ideas that we are pursuing in the longer term that may significantly impact the core user experience of the Kubernetes provider. Research topics are discovery only and are not guaranteed to be included in a future release.
+
+### Kubernetes Manifest Resource
+
+Issue: [#215](https://github.com/hashicorp/terraform-provider-kubernetes/issues/215)
+
+We’ve been evaluating the best way to support custom resources in the Kubernetes provider. The [initial work](https://github.com/hashicorp/terraform-provider-kubernetes-alpha) led us to create a new provider with a single resource, called `kubernetes_manifest`. We did this because the functionallity needed to accomidate the dynamic schema of custom resources wasn't available in the Terraform SDK.
```
Comment (aareet) [inline _about/ROADMAP.md:56]: typo - `functionality needed to accommodate`

File: README.md [hunk 4]
```diff
@@ -11,6 +11,7 @@
   - [Examples](https://github.com/hashicorp/terraform-provider-kubernetes/tree/master/_examples)
 - Mailing list: [Google Groups](http://groups.google.com/group/terraform-tool)
 - Chat: [#terraform-providers in Kubernetes](https://kubernetes.slack.com/messages/CJY6ATQH4) ([Sign up here](http://slack.k8s.io/))
+- Roadmap: [Q3 2020](_about/ROADMAP.md)
```
Comment (aareet) [inline README.md:14]: As a more general question - will we have separate roadmap documents for the helm, the operator and the kubernetes provider? If not, may make sense to centralize the efforts here and place operator work in `discovery`

File: _about/ROADMAP.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
+# Q3 2020 Roadmap
+
+Each quarter the team will highlight areas of focus for our work and upcoming research.
+ 
+We select items to include in the roadmap from community issues and internal priorities. When community pull requests exist for an item, we will prioritize working with the original authors to include their contributions. If the author can no longer take on the implementation, HashiCorp may complete any additional work needed. 
+
+Each release will include necessary tasks that lead to the completion of the stated goals as well as community pull requests, enhancements, and features that are not highlighted in the roadmap. 
+
+To make contribution easier, we’ll be using the [`Help Wanted`](https://github.com/hashicorp/terraform-provider-kubernetes/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) tag to point to issues we’d like to include in this quarter’s series of releases.
```
Comment (aareet) [inline _about/ROADMAP.md:9]: Can you also link out to the contribution guide for people who see this document and want to pick up something in help-wanted?

File: _about/ROADMAP.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+Each quarter the team will highlight areas of focus for our work and upcoming research.
+ 
+We select items to include in the roadmap from community issues and internal priorities. When community pull requests exist for an item, we will prioritize working with the original authors to include their contributions. If the author can no longer take on the implementation, HashiCorp may complete any additional work needed. 
+
+Each release will include necessary tasks that lead to the completion of the stated goals as well as community pull requests, enhancements, and features that are not highlighted in the roadmap. 
+
+To make contribution easier, we’ll be using the [`Help Wanted`](https://github.com/hashicorp/terraform-provider-kubernetes/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) tag to point to issues we’d like to include in this quarter’s series of releases. 
+
+This quarter (July-September ‘20) we will prioritze the following areas of work:
```
Comment (aareet) [inline _about/ROADMAP.md:11]: `prioritize`

File: _about/ROADMAP.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+## Currently In Progress
+
+### 2.0.0 
+
+Major version releases include code removals, deprecations, and breaking changes. A corresponding “upgrade guide” will be published alongside the release. 
+
+In the 2.0.0 release we'll focus on three areas:
+ - Upgrading to [Terraform Plugin SDK v2](https://www.terraform.io/docs/extend/guides/v2-upgrade-guide.html)
+ - [Removing support for Terraform 0.11](https://github.com/hashicorp/terraform-provider-kubernetes/issues/905)
+ - [Changing provider credentialling]
```
Comment (aareet) [inline _about/ROADMAP.md:22]: I think we could provide more context here on what we're trying to improve with these changes

File: _about/ROADMAP.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+### Coverage
```
Comment (aareet) [inline _about/ROADMAP.md:24]: I think we could add some context here to describe how we're planning to approach improvements according to these themes. Could just be a line like "In an attempt to focus on specific areas of the user experience when improving the provider, we plan to identify and focus on a set of themes each quarter. Here are the 3 themes we plan to work on this quarter." or something like that?

File: _about/ROADMAP.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+### Lifecycle
+
+Issues: [theme/lifecycle](https://github.com/hashicorp/terraform-provider-kubernetes/issues?q=is%3Aopen+is%3Aissue+label%3Atheme%2Flifecycle)
+
+Goal: Lifecycle improvements to match user workflows
+
+There are several requests to modify behavior based on Kubernetes lifecycle events, for example, waiting or not waiting for a pod to become ready before a deployment is considered complete. We’ll look at this and other areas where additional lifecycle controls make sense in the provider. 
+
+### Errors
```
Comment (aareet) [inline _about/ROADMAP.md:40]: What about designating this as a stretch goal given we have 10 weeks left in the quarter? There's going to be some collateral unnamed work around 0.13 compatibility, SDK upgrades, theme/diffs and the test infrastructure improvements we're talking about. Happy to discuss offline to decide on this.

File: README.md (hunk 4 above)
Comment (Redeux) [reply README.md:14]: Originally I thought I would combine them into a single document, but now I think they should all have their own artifact.  It probably still makes sense to reference the operator in this roadmap for visibility.

File: _about/ROADMAP.md (hunk 2 above)
Comment (Redeux) [reply _about/ROADMAP.md:58]: I need a better spell check workflow for my markdown docs, clearly.

File: _about/ROADMAP.md (hunk 5 above)
Comment (Redeux) [reply _about/ROADMAP.md:9]: Great idea

File: _about/ROADMAP.md (hunk 8 above)
Comment (Redeux) [reply _about/ROADMAP.md:24]: Good idea.  Let me know if you like the wording I added.

File: _about/ROADMAP.md (hunk 9 above)
Comment (Redeux) [reply _about/ROADMAP.md:40]: Sounds reasonable

**Pull request merged**